my-webxr-app/src/data/DataLayer.ts

```diff
@@ -0,0 +1,131 @@
+import DataAbstractor from './DataAbstractor';
+import { Repository } from '../repository/Repository';
+import DbRepository from '../repository/DbRepository';
+import DataPoint from '../repository/DataPoint';
+
+/**
+ * The Data Layer provides a set of methods for working with CSV and PCA data.
+ */
+export default class DataLayer implements DataAbstractor {
+  // @ts-expect-error temp ignore
+  private repository: Repository;
+
+  /**
+   * Create a new Data Layer instance.
+   * @param dbName (optional) the name of the Data Repository.
+   */
+  constructor(dbName?: string) {
+    this.repository = new DbRepository(dbName ?? 'DAL_DB');
+  }
+
+  /**
+   * Transport a stream of batched data to be referenced column-wise instead of row-wise.
+   * @param batchItems A 2D array of CSV data that is referenced by row.
+   * @protected
+   */
+  protected static transposeData(batchItems: BatchedDataStream) {
+    const rows = batchItems.length;
+    const cols = batchItems[0]?.length ?? 0;
+    const grid: BatchedDataStream = [];
+
+    for (let j = 0; j < cols; j += 1) {
+      grid[j] = Array(rows);
+    }
+
+    for (let i = 0; i < rows; i += 1) {
+      for (let j = 0; j < cols; j += 1) {
+        grid[j][i] = batchItems[i][j];
+      }
+    }
+
+    return grid;
+  }
+
+  /**
+   * Calculate the statistics for a set of data. Items that are not a number are treated as 0.
+   * @param batchItems transposed data referenced column-wise.
+   * @protected
+   */
+  protected static calculateStatistics(batchItems: BatchedDataStream) {
+    const statsArray: Array<ColumnStatistics> = [];
+
+    batchItems.forEach((column) => {
+      // Convert non-number types into 0 for the stats calculations.
```

Comment:
What would be classified as non-number types?

Reply:
From line 123 below: `export type BatchedDataStream = Array<Array<string | number | null>>;`. This is based on the type definition from the CSV reader output. 

There could be instances where the data contains `string` or `null` values. These get converted to 0 in order to avoid breaking the statistics. We decided upon this method in our group pairing last night.

Reply:
Got it, that sounds good to me!